### PR TITLE
mavutil.py: support multiple clients with a timeout on udpin endpoints

### PR DESCRIPTION
**Pull request by Williangalvani**
This allows us to have systems connecting only when required. After 3 seconds with no communication, the clients get "disconnected".
We never disconnect the last client, so this won't break existent systems that don't communicate for long periods of time.

### CODE REVIEW

File: mavutil.py [hunk 1]
```diff
@@ -898,15 +902,25 @@ def recv(self,n=None):
             if e.errno in [ errno.EAGAIN, errno.EWOULDBLOCK, errno.ECONNREFUSED ]:
                 return ""
             raise
-        if self.udp_server or self.broadcast:
+        if self.udp_server:
+            self.clients.add(new_addr)
+            self.clients_last_alive[new_addr] = time.time()
+        elif self.broadcast:
             self.last_address = new_addr
```
Comment (patrickelectric) [inline mavutil.py:909]: Should we do the same thing for broadcast ?

File: mavutil.py [hunk 2]
```diff
@@ -898,15 +902,25 @@ def recv(self,n=None):
...
 
     def write(self, buf):
         try:
             if self.udp_server:
-                if self.last_address:
-                    self.port.sendto(buf, self.last_address)
+                current_time = time.time()
+                to_remove = set()
+                for address in self.clients:
+                    if current_time - self.clients_last_alive[address] <= self.timeout:
```
Comment (jaxxzer) [inline mavutil.py]: it's a better habit to add times instead of subtracting them to deal with overflows.

if last alive + self.timeout < current_time...

Comment (jaxxzer) [reply mavutil.py]: (probably won't matter in python, but the habit is the point)

File: mavutil.py (hunk 1 above)
Comment (jaxxzer) [reply mavutil.py:909]: has this been resolved?

Comment (Williangalvani) [reply mavutil.py:909]: I don't think we need it right now. There's some additional logic in the broadcast section I didn't quite understand the reasoning behind, too.

**Review by jaxxzer — APPROVED** [review]

**Pull request merged**